unittests/resources/checks_unlisted/required_versions.py

```diff
@@ -0,0 +1,30 @@
+import reframe as rfm
+
+
+@rfm.simple_test
+class Test1Check(rfm.RegressionTest):
+    pass
+
+
+@rfm.required_version()
+@rfm.simple_test
+class Test2Check(rfm.RegressionTest):
+    pass
+
+
+@rfm.simple_test
+@rfm.required_version('>100.0')
+class Test3Check(rfm.RegressionTest):
+    pass
+
+
+@rfm.required_version('1.0..2.0', '<100.0')
+@rfm.simple_test
+class Test4Check(rfm.RegressionTest):
+    pass
+
+
+@rfm.simple_test
+@rfm.required_version('!=2.0')
+class Test5Check(rfm.RegressionTest):
+    pass
```

Comment:
I don't think we need this file at all. We already have the `unittests/resources/checks_unlisted/good.py` that is being loaded by the check loader. We could simply decorate one of the tests in there with a `required_version` that is always true and add another test, which should be skipped.